docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+const { InMemoryCache } = require("apollo-cache-inmemory")
+const { split } = require("apollo-link")
+const { HttpLink } = require("apollo-link-http")
+const { WebSocketLink } = require("apollo-link-ws")
+const { getMainDefinition } = require("apollo-utilities")
+const fetch = require("node-fetch")
+const gql = require("graphql-tag")
+const WebSocket = require("ws")
+// highlight-end
+
+const POST_NODE_TYPE = `Post`
+
+exports.sourceNodes = async ({
+// ...
+```
+
+Then you can copy this code that sets up the necessary pieces of the Apollo client and paste it after your imports:
+
+```javascript:title=source-plugin/gatsby-node.js
+// ... imports
+const WebSocket = require("ws")
+
+const POST_NODE_TYPE = `Post`
+
+// highlight-start
+const client = new ApolloClient({
+  link: split(
+    ({ query }) => {
+      const definition = getMainDefinition(query)
+      return (
+        definition.kind === "OperationDefinition" &&
+        definition.operation === "subscription"
+      )
+    },
+    new WebSocketLink({
+      uri: `ws://localhost:4000`, // or `ws://gatsby-source-plugin-api.glitch.me/`
+      options: {
+        reconnect: true,
+      },
+      webSocketImpl: WebSocket,
+    }),
+    new HttpLink({
+      uri: "http://localhost:4000", // or `https://gatsby-source-plugin-api.glitch.me/`
+      fetch,
+    })
+  ),
+  cache: new InMemoryCache(),
+})
+// highlight-end
+
+exports.sourceNodes = async ({
+// ...
+```
+
+You can read about each of the packages that are working together in [Apollo's docs](https://www.apollographql.com/docs/react/), but the end result is creating a `client` that you can use to call methods like `query` on to get data from the source it's configured to work with. In this case, that is `localhost:4000` where you should have the API running. If you can't configure the API to run locally, you can update the URLs for the client to use `gatsby-source-plugin-api.glitch.me` where a version of the API is deployed, instead of `localhost:4000`.
+
+#### Query data from the API
+
+Now you can replace the hardcoded data in the `sourceNodes` function with a GraphQL query:
+
+```diff:title=source-plugin/gatsby-node.js
+  exports.sourceNodes = async ({
+    actions,
+    createContentDigest,
+    createNodeId,
+    getNodesByType,
+  }) => {
+    const { createNode, touchNode, deleteNode } = actions
+
+-   const data = {
+-     posts: [
+-       { id: 1, description: `Hello world!` },
+-       { id: 2, description: `Second post!` },
+-     ],
+-   }
++   const { data } = await client.query({
++     query: gql`
++       query {
++         posts {
++           id
++           description
++         }
++       }
++     `,
++   })
+
+    // ...
+```
+
+Now you're creating nodes based on data coming from the API, neat! However, only the `id` and `description` fields are coming back from the API and being saved to each node, so add the rest of the fields to the query so that there will be the same data available to Gatsby.
```

Comment:
```suggestion
Now you're creating nodes based on data coming from the API, neat! However, only the `id` and `description` fields are coming back from the API and being saved to each node, so add the rest of the fields to the query so that the same data is available to Gatsby.
```